ICAFIX/hcp_fix

```diff
@@ -337,12 +401,24 @@ fi
 log_Msg "using training data file: ${TrainingData}"
 
 # set up fix command
+# use array for whitespace safety, even if the rest of the script isn't
 if [[ ${doMotionRegression} == "TRUE" ]]; then
-    #use array for whitespace safety, even if the rest of the script isn't
     fix_cmd=("${FSL_FIXDIR}/fix" "${fmrihp}.ica" "${TrainingData}" "${FixThresh}" -m -h "${hp}")
 else
-    #-h is actually a subargument to -m, and will cause problems if specified without (or even not directly following) -m
-    fix_cmd=("${FSL_FIXDIR}/fix" "${fmrihp}.ica" "${TrainingData}" "${FixThresh}")
+	old_or_new_fix=$(determine_old_or_new_fix)
+	if [[ "${old_or_new_fix}" == "NEW" ]]; then
+		# In the "NEW" (1.06.12 or later) version of FIX, -h is no longer a subargument to -m, and can occur by itself
+		fix_cmd=("${FSL_FIXDIR}/fix" "${fmrihp}.ica" "${TrainingData}" "${FixThresh}" -h "${hp}")
+
+	elif (( hp < 0 )); then
+		# Ok to proceed regardless of FIX version, since not filtering, so we don't need the filtering of the CIFTI
+		# that occurs in fix_3_clean. Simply don't supply either the -m or -h flags.
+		fix_cmd=("${FSL_FIXDIR}/fix" "${fmrihp}.ica" "${TrainingData}" "${FixThresh}")
+
+	else  # "OLD" and (hp >= 0)
+		# Already aborted in this case earlier, but include again here for completeness
+		log_Err_Abort "Due to a restriction in the FIX argument parsing, do_motion_regression=$3 is not supported for FIX versions prior to 1.06.12"
```

Comment:
We will already abort when using old fix and `hp` is enabled, so we don't need to check old or new again here.  If we test only `hp`, we can issue calls that will work in all supported cases, and even if we didn't correctly abort in advance, the correct call for new fix would simply fail for old fix anyway.

Reply:
I did it that way so that we could parallel the `doMotionRegression=TRUE` case and still pass `-h -1` as an argument to fix if it is the "NEW" fix version and hp<0.  Similarly, in the MR-FIX script, we pass `-m -h -1` to fix in the `doMotionRegression=TRUE` case rather than just `-m`.

Reply:
i.e., this way keeps the calls to fix identical to what we were using previously (which seems preferable to me), in which case we need to add a call specific to the NEW fix version to allow `-h` without `-m`.  The repeated abort is indeed redundant, but I felt an explanation of what occurs in that situation is needed anyway given that the abort of that condition happens much earlier in the code.

Reply:
And, to be entirely technical about it `-m` (by itself) and `-m -h -1` aren't strictly the same thing, since `fix` itself checks for a design.fsf file if `-h` isn't specified, and if it finds a design.fsf it then pulls the filter value out of that.  So, the way I implemented it _guarantees_ 100% backwards compatibility without any potential weird gotchas.